### ISSUE TITLE
[HttpKernel] #[MapQueryString] not nullable arguments

### ISSUE THREAD

**Issue by plozmun** [symfony/symfony#50344]
### Symfony version(s) affected

6.3.0-BETA3

### Description

Given a DTO 
```php
<?php

namespace App\DTO;

class QueryStringDTO
{
    public function __construct(
        public ?string $test
    ) {
    }
}
```
And a controller  
```
    #[Route("test")]
    public function __invoke(#[MapQueryString] QueryStringDTO $query)
    {

    }
```
Every works with route 
/test?test=1 
but 404 response with only /test

It's seems that $payload is null and metadata argument is not nullable

https://github.com/symfony/symfony/blob/6.3/src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php#L133

### How to reproduce

Reproducer: 

https://github.com/plozmun/mapQueryString-not-nullable

**Comment by stof** [comment]
> Every works with route
> /test?test=1
> but 404 response with only /test

Well, your controller argument is required and the query string parameter is not provided. So the 404 is expected to me.

**Comment by nicolas-grekas** [comment]
@stof is right. Maybe you meant #50339?

**Comment by IceMaD** [comment]
I encountered the same problem today.
My use case is a list API with filters which are all optional.

I understand the answer given, but it seems incorrect to me.
If the DTO allows the parameter to be null, then the "no query string" behavior is supported by the DTO.
It shouldn't be up to the resolver to make the decision to return a 404.

I don't know enough of the code to know if there are any other implications,
but I tried out of curiosity to remove the condition in `\Symfony\Component\HttpKernel\Controller\ArgumentResolver\RequestPayloadValueResolver::mapQueryString`.

The result is the following:
- The query is valid if the DTO does not require any parameters and its fields are null
- A 404 is triggered if the DTO requires at least one parameter
